src/App.js

```diff
@@ -14,11 +14,96 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: goodsFromServer,
+    isVissible: false,
+    isReversed: false,
+    sortBy: '',
+  };
+
+  showList = () => {
+    this.setState({ isVissible: true });
+  };
+
+  reverseList = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  sortAsString = () => {
+    this.setState({ sortBy: 'string' });
+  };
```

Comment:
Probably it would be better to name it smth like 'sortByAlphabet' (more meaningful)